File: Aditya-Pratap/Sorting/910.py
class Solution:
    def smallestRangeII(self, nums: List[int], k: int) -> int:
        nums.sort()
        n=len(nums)
        ans=nums[n-1]-nums[0]
        left=nums[0]+k
        right=nums[n-1]-k
        for i in range(n-1):
            mini=min(left,nums[i+1]-k)
            maxn=max(right,nums[i]+k)
            ans=min(ans,maxn-mini)
        return ans

File: [Manya]/Leetcode/Sorting/910.java
package Sorting;

import java.util.Arrays;

class Solution {
    public int smallestRangeII(int[] nums, int k) {
        Arrays.sort(nums);

        int bestL = nums[0] + k;
        int bestR = nums[nums.length - 1] - k;
        int result = nums[nums.length - 1] - nums[0];

        for(int i = 0; i < nums.length - 1; i++){
            int min = Math.min(nums[i+1] - k , bestL);
            int max = Math.max(nums[i] + k , bestR);
            result = Math.min(result, max-min);
        }
        return result;
    }
}